src/lib.rs

```diff
@@ -61,7 +94,15 @@ use std::path::{Path, PathBuf};
 /// }
 /// ```
 pub fn home_dir() -> Option<PathBuf> {
-    home_dir_inner()
+    home_dir_from(&OS_ENV)
+}
+
+/// Variant of home_dir where the environment source is parameterised. This is
+/// specifically to support in-process testing scenarios as environment
+/// variables and user home metadata are normally process global state. See the
+/// OsEnv trait.
```

Comment:
The doc here could be rephrased a little bit:
```suggestion
/// Returns the path of the current user's home directory from [`Env::home_dir`].
```